File: 1023-camelcase-matching/1023-camelcase-matching.py
class Solution:
    def camelMatch(self, queries: List[str], pattern: str) -> List[bool]:
        def check(s, pattern):
            j, n = 0, len(pattern)
            for ch in s:
                if j < n and ch == pattern[j]:
                    j += 1
                elif ch == ch.lower():
                    continue
                else:
                    return False
            return j == n
          
        
        return [check(q, pattern) for q in queries]

File: 1023-camelcase-matching/1023-camelcase-matching.js
/**
 * @param {string[]} queries
 * @param {string} pattern
 * @return {boolean[]}
 */
var camelMatch = function(queries, pattern) {
  return queries.map((query) => isMatched(query, pattern));
};

const isMatched = (str, pattern) => {
  let j = 0;
  
  for (let ch of str) {
    if (ch == pattern[j]) j++;
    else if (ch === ch.toLowerCase()) continue;
    else return false;
  }
  
  return j === pattern.length;
}